### PR TITLE
Adopt `PluginDirectoryServiceRemote` changes

### PR DESCRIPTION
**Pull request by crazytonyli**
> [!Note]
> This PR is built on top of #22612.

This PR updates the app code to accept the `PluginDirectoryServiceRemote` breaking changes introduced in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/725.

## Test Instructions

1. Log into a self hosted site
2. Go to "Plugins"
3. Verify the Plugins screen can be loaded
4. Verify plugin details can be viewed.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the "Test Instructions"

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist: N/A

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mokagio — APPROVED** [review]
Approving to unblock while waiting for the Kit reference to be updated to the merge commit of the PR into `trunk`.


I installed the Jetpack App Center build and verified plugins worked.

<table>
<tr>
 <td>

IMG_A1741D133F8B-1

</td>
 <td>

IMG_A1741D133F8B-2

</td>
</table>

File: Podfile
Comment (mokagio) [inline Podfile:54]: This commit points to https://github.com/wordpress-mobile/WordPressKit-iOS/pull/727 which, I'm happy to say, seems to address the reader issue that the UI tests picked up in previous runs (see #22612)

**Comment by crazytonyli** [comment]
Changes in this PR have been moved to https://github.com/wordpress-mobile/WordPress-iOS/pull/22612

**Pull request closed without merging**
